src/main/java/net/neoforged/neoforge/transfer/handlers/wrappers/RangedResourceHandler.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.handlers.wrappers;
+
+import com.google.common.base.Preconditions;
+import java.util.Objects;
+import java.util.function.Supplier;
+import net.neoforged.neoforge.transfer.handlers.resources.IResourceHandler;
+import net.neoforged.neoforge.transfer.resources.IResource;
+import net.neoforged.neoforge.transfer.transaction.TransactionContext;
+
+/**
+ * A wrapper that delegates all calls to a range of indices of a handler.
+ * <p>
+ * <b>By itself, this does not handle snapshotting.</b> It is expected the delegated handlers take care of what needs to be journaled.
+ *
+ * @param <T> The type of resource this handler manages.
+ */
+public class RangedResourceHandler<T extends IResource> extends DelegatingResourceHandler<T> {
+    protected int start;
+    protected int end;
+
+    public RangedResourceHandler(IResourceHandler<T> delegate, int start, int end) {
+        this(() -> delegate, start, end);
+    }
+
+    public RangedResourceHandler(Supplier<IResourceHandler<T>> delegate, int start, int end) {
```

Comment:
This needs docs, especially to convey that start is inclusive, but end is exclusive